modules/apps/transfer/types/forwarding.go

```diff
@@ -8,26 +8,52 @@ import (
 
 const MaximumNumberOfForwardingHops = 16 // denotes the maximum number of forwarding hops allowed
 
-// NewForwarding creates a new Forwarding instance given a memo and a variable number of hops.
-func NewForwarding(memo string, hops ...Hop) Forwarding {
+// NewForwarding creates a new Forwarding instance given an unwind value and a variable number of hops.
+func NewForwarding(unwind bool, hops ...Hop) Forwarding {
 	return Forwarding{
-		Memo: memo,
-		Hops: hops,
+		Unwind: unwind,
+		Hops:   hops,
 	}
 }
 
 // Validate performs a basic validation of the Forwarding fields.
 func (fi Forwarding) Validate() error {
 	if len(fi.Hops) > MaximumNumberOfForwardingHops {
-		return errorsmod.Wrapf(ErrInvalidForwarding, "number of hops in forwarding path cannot exceed %d", MaximumNumberOfForwardingHops)
+		return errorsmod.Wrapf(ErrInvalidForwarding, "number of hops in forwarding packet data cannot exceed %d", MaximumNumberOfForwardingHops)
 	}
 
-	if len(fi.Memo) > MaximumMemoLength {
+	for _, hop := range fi.Hops {
+		if err := host.PortIdentifierValidator(hop.PortId); err != nil {
+			return errorsmod.Wrapf(err, "invalid source port ID %s", hop.PortId)
```

Comment:
To differentiate from the same error string that could happen if `MsgTransfer`'s `SourcePort` or `SourcePort` is invalid:

```suggestion
			return errorsmod.Wrapf(err, "invalid hop source port ID %s", hop.PortId)
```